dvc/stage/__init__.py

```diff
@@ -60,21 +60,60 @@ def loads_from(cls, repo, path, wdir, data):
 
 
 def create_stage(cls, repo, path, external=False, **kwargs):
+    """Create a stage from the class *cls*
+
+    :param cls: the class from which the stage will be
+                created (e.g Stage, PipelineStage).
+    :param repo: the repository object (Repo class).
+    :param path: the path to the dvc filename. If None,
+                 path become <out>.dvc
+    :param external: if True, prevents users from accidentally
+                     using external outputs.
+                     See https://github.com/iterative/dvc/issues/1545
+                     for more details.
+
+    the method can also receive other params as:
+    :param outs: a list of output file (or dir) names.
+    :param deps: a list of urls that represents the source of files.
+    # NOTE there are other possible parameters.
+
+    :returns: a *cls* object representing the stage.
+    """
     from dvc.dvcfile import check_dvc_filename
 
     wdir = os.path.abspath(kwargs.get("wdir", None) or os.curdir)
-    path = os.path.abspath(path)
-    check_dvc_filename(path)
     check_stage_path(repo, wdir, is_wdir=kwargs.get("wdir"))
+
+    # path is None because we define its filename from the dependency.
```

Comment:
@julsal please, correct me if I'm wrong here. My understanding is that we do this (the `get_file_name` approach) only to cover the case when we pass `gdrive://ID` to the `dvc import-url` since we don't know the file name in advance?

Reply:
Yes, that's correct.